### PR TITLE
[stable/mysql] Accept custom mysql config files

### PR DESCRIPTION
**Pull request by fabiormoura**
The official MySQL image allows passing custom MySQL config files but the helm chart was not exposing this as a first class feature.

### CODE REVIEW

File: stable/mysql/README.md [hunk 1]
```diff
@@ -88,3 +89,18 @@ By default a PersistentVolumeClaim is created and mounted into that directory. I
 you can change the values.yaml to disable persistence and use an emptyDir instead.
 
 > *"An emptyDir volume is first created when a Pod is assigned to a Node, and exists as long as that Pod is running on that node. When a Pod is removed from a node for any reason, the data in the emptyDir is deleted forever."*
+
+## Custom MySQL configuration files
+
+The [MySQL](https://hub.docker.com/_/mysql/) image accepts custom configuration files at the path `/etc/mysql/conf.d`. If you want to use a customized MySQL configuration, you can create your alternative configuration files by passing the file contents on the `configurationFiles` attribute. Note that according with the MySQL documentation only files ending with `.cfn` are loaded.
```
Comment (dhilipkumars) [inline stable/mysql/README.md]: `Note that according with the MySQL documentation only files ending with`

Should be `according to MySQL documentation`

File: stable/mysql/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{{- if .Values.configurationFiles }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "fullname" . }}
+data:
+{{- range $key, $val := .Values.configurationFiles }}
+  {{ $key }}: |-
```
Comment (dhilipkumars) [inline stable/mysql/templates/configmap.yaml:8]: Just for clarification: If the actual value already as the `|-` _(as mentioned in the example)_ would this not create a problem?

Comment (fabiormoura) [reply stable/mysql/templates/configmap.yaml:8]: The file gets rendered correctly even with those characters. I verified and deployed the chart on my testing k8s cluster using the same configuration as in the README file and it works fine.

File: stable/mysql/README.md (hunk 1 above)
Comment (fabiormoura) [reply stable/mysql/README.md]: Fixed.

**Review by dhilipkumars — APPROVED** [review]

**Comment by dhilipkumars** [comment]
LGTM from my side
cc: @kubernetes/charts-maintainers PTAL and approve.

**Comment by davidassetowl** [comment]
Configmap is not generated:

`
bash-3.2$ helm install --name db-common -f charts/values-mysql.yaml stable/mysql
NAME:   db-common
LAST DEPLOYED: Wed Aug 23 10:32:35 2017
NAMESPACE: localhost
STATUS: DEPLOYED

RESOURCES:
==> v1/Service
NAME             CLUSTER-IP  EXTERNAL-IP  PORT(S)   AGE
db-common-mysql  10.0.0.238  <none>       3306/TCP  1s

==> v1beta1/Deployment
NAME             DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
db-common-mysql  1        1        1           0          1s

==> v1/Secret
NAME             TYPE    DATA  AGE
db-common-mysql  Opaque  2     1s

==> v1/PersistentVolumeClaim
NAME             STATUS  VOLUME                                    CAPACITY  ACCESSMODES  STORAGECLASS  AGE
db-common-mysql  Bound   pvc-8f207cd6-879a-11e7-9a0b-080027a55976  8Gi       RWO          standard      1s
`

issue raised: [https://github.com/kubernetes/charts/issues/1816](#1816)

**Comment by davidassetowl** [comment]
Configmap is not generated:

`bash-3.2$ helm install --name db-common -f charts/values-mysql.yaml stable/mysql
NAME:   db-common
LAST DEPLOYED: Wed Aug 23 10:32:35 2017
NAMESPACE: localhost
STATUS: DEPLOYED

RESOURCES:
==> v1/Service
NAME             CLUSTER-IP  EXTERNAL-IP  PORT(S)   AGE
db-common-mysql  10.0.0.238  <none>       3306/TCP  1s

==> v1beta1/Deployment
NAME             DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
db-common-mysql  1        1        1           0          1s

==> v1/Secret
NAME             TYPE    DATA  AGE
db-common-mysql  Opaque  2     1s

==> v1/PersistentVolumeClaim
NAME             STATUS  VOLUME                                    CAPACITY  ACCESSMODES  STORAGECLASS  AGE
db-common-mysql  Bound   pvc-8f207cd6-879a-11e7-9a0b-080027a55976  8Gi       RWO          standard      1s`

issue raised: [https://github.com/kubernetes/charts/issues/1816](#1816)

**Comment by davidassetowl** [comment]
Configmap is not generated:

```
bash-3.2$ helm install --name db-common -f charts/values-mysql.yaml stable/mysql
NAME:   db-common
LAST DEPLOYED: Wed Aug 23 10:32:35 2017
NAMESPACE: localhost
STATUS: DEPLOYED

RESOURCES:
==> v1/Service
NAME             CLUSTER-IP  EXTERNAL-IP  PORT(S)   AGE
db-common-mysql  10.0.0.238  <none>       3306/TCP  1s

==> v1beta1/Deployment
NAME             DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
db-common-mysql  1        1        1           0          1s

==> v1/Secret
NAME             TYPE    DATA  AGE
db-common-mysql  Opaque  2     1s

==> v1/PersistentVolumeClaim
NAME             STATUS  VOLUME                                    CAPACITY  ACCESSMODES  STORAGECLASS  AGE
db-common-mysql  Bound   pvc-8f207cd6-879a-11e7-9a0b-080027a55976  8Gi       RWO          standard      1s
```

issue raised:  #1816

**Pull request merged**